Replace d3-voronoi with d3-delaunay.

domoritz opened the pull request:
Fixes #1492

jheer commented:
Thanks for the PR! The failing tests are due to the test scene for the airports example, where the voronoi polygon path strings have changed.

I don't want to merge until we have a solution in place for d3/d3-delaunay#19. Vega can certainly encounter voronoi instances with less than 3 input points.

domoritz commented:
Yeah, I agree that https://github.com/d3/d3-delaunay/issues/19 is a blocker for us. I wonder whether @mourner or @mbostock have this issue on their queue.

domoritz commented:
There is also https://github.com/d3/d3-delaunay/issues/20 but it's less of an issue.

jheer commented:
Hmm, I think d3/d3-delaunay#20 is absolutely a blocker as well. I've seen Vega users generate collinear points for a Voronoi diagram to enable a form of "nearest" selection.

domoritz commented:
d3/d3-delaunay#20 has been fixed

domoritz commented:
@jheer I updated the PR and fixed any remaining issues. Can you rebuild the airport example? I tried it locally and tests are passing but the results differ from what's happening on Travis.

domoritz commented:
Ahh, it's because of #1893

mbostock commented on packages/vega-voronoi/src/Voronoi.js:
```diff
@@ -22,25 +22,23 @@ Voronoi.Definition = {
-
-  this.value = (diagram = diagram(data));
+  delaunay = Delaunay.from(data, _.x, _.y);
+  extent = _.size ? [0, 0 , ..._.size] : _.extent ? [..._.extent[0], _.extent[1][0]] : defaultExtent;
```
I think you’ll want `[..._.extent[0], ..._.extent[1]]` here?

domoritz replied:
Yes. Not sure what my past me was thinking here.

domoritz commented:
I updated to the latest version of d3-delaunay and fixed the build. This is ready to go (squashed).

jheer commented:
Why did you merge? I am working on this... Ugh.

domoritz commented:
Ups, I took the approval as a permission to merge. I squashed all of the changes into one commit so hopefully it's not too confusing.
